packages/external-db-config/lib/readers/aws_config_reader.js

```diff
@@ -23,77 +21,85 @@ class AwsConfigReader {
+    const externalConfig = await this.readExternalConfig().catch(emptyConfig)
+    const { host, username, password, DB, SECRET_KEY, HOST, PASSWORD, USER } = { ...process.env, ...externalConfig }
+    const config = {  host: host || HOST, username: username || USER, password: password || PASSWORD, DB, SECRET_KEY }
+    return { config, secretMangerError: externalConfig.secretMangerError }
+  }
+
   async validate() {
-    try {
-      const cfg = await this.readExternalConfig()
-      return { missingRequiredSecretsKeys: checkRequiredKeys(cfg, ['host', 'username', 'password', 'DB', 'SECRET_KEY']) }
-    } catch (err) {
-      return { configReadError: err.message, missingRequiredSecretsKeys: [] }
-    }
+      const { config, secretMangerError } = await this.getExternalAndLocalEnvs()
+      return { missingRequiredSecretsKeys: checkRequiredKeys(config, ['host', 'username', 'password', 'DB', 'SECRET_KEY']), secretMangerError }
   }
 }
 
-  class AwsDynamoConfigReader {
-    constructor(region, secretId) {
-      this.region = region
-      this.secretId = secretId
-     }
-
-     async readExternalConfig() {
-      const client = new SecretsManagerClient({ region: this.region })
-      const data = await client.send(new GetSecretValueCommand({ SecretId: this.secretId }))
-      return JSON.parse(data.SecretString)
+class AwsDynamoConfigReader {
+  constructor(region, secretId) {
+    this.region = region
+    this.secretId = secretId
     }
 
     async readConfig() {
+      const { config } = await this.getExternalAndLocalEnvs()
       if (process.env.NODE_ENV === 'test') {
-        return { region: this.region, secretKey: process.env.SECRET_KEY, endpoint: process.env.ENDPOINT_URL }
+        return { region: this.region, secretKey: config.SECRET_KEY, endpoint: process.env.ENDPOINT_URL }
       }
-      const cfg = await this.readExternalConfig()
-                            .catch(() => (EmptyConfig))
-  
-      return { region: this.region, secretKey: cfg.SECRET_KEY }
+      return { region: this.region, secretKey: config.SECRET_KEY }
     }
+    
+    async getExternalAndLocalEnvs() { 
+      const externalConfig = await this.readExternalConfig().catch(emptyConfig)
+      const { SECRET_KEY } = { ...process.env, ...externalConfig }
+      const config = { SECRET_KEY }
 
-    async validate() {
-      try {
-        const cfg = process.env.NODE_ENV !== 'test' ? await this.readExternalConfig() : process.env
-        return { missingRequiredSecretsKeys: checkRequiredKeys(cfg, ['SECRET_KEY']) }
-        
-      } catch (err) {
-        return { configReadError: err.message, missingRequiredSecretsKeys: ['SECRET_KEY'] }
-      }
+      return { config, secretMangerError: externalConfig.secretMangerError }
     }
+
+    async readExternalConfig() {
+    const client = new SecretsManagerClient({ region: this.region })
+    const data = await client.send(new GetSecretValueCommand({ SecretId: this.secretId }))
+    return JSON.parse(data.SecretString)
```

Comment:
If the secret reading was successful, return an object with a `externalConfig` property